plugin.lua

```diff
@@ -70,16 +33,16 @@ local on_event = require("factorio-plugin.on-event")
 ---@param text string @ The content of file
 ---@return nil|Diff[]
 function OnSetText(uri, text)
+
+  ---Not sure about this one if files are in the library it will not process
+  ---cross workspace requires seem to require the path in the library?
+  --TODO report that as sumneko bug? Needs more testing, maybe individual settings.
```

Comment:
raiguard and I have tested this yesterday and here's how it behaves:
`require`ing a file that is in a symlink within a library simply doesn't work, ever.
`require`ing a file using the `__modname__` prefix only works about 1/5 times on first load, but works 100% of the time when modifying the file with the `__modname__` require(s) in it. This appears to be some kind of race condition in `sumneko.lua` involving `require` resultion and plugin diffing.
(At least I'm assuming that's what you're talking about here, I'm not actually sure)

Reply:
Somewhat
For the race condition you can see it with events if the file containing script.on_events is open sometimes after load/reload it won't update until you start typing.

`__modname__` requires seem to only look in the current scope,   if `__modname__` is your current scope and runtime.path includes your mods folder then magic happens
if `__modname__` is another workspace folder (and the path is set as above)  it will only read `__modname__` if it is included in workspace.library

What I have not tested yet was if  `__modname__` is the current scope and `__modname__` is in the library path does it get skipped with this check.